File: output/gemma4-e4b/avatar/Python/Java/codeforces_612_B.java
import java.util.Scanner;
import java.lang.Math;

public class codeforces_612_B {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        // Read N
        if (!scanner.hasNextInt()) {
            return;
        }
        int n = scanner.nextInt();

        // Read array A
        int[] a = new int[n];
        for (int i = 0; i < n; i++) {
            if (!scanner.hasNextInt()) {
                // Handle case where input might end prematurely
                return;
            }
            a[i] = scanner.nextInt();
        }

        // Initialize array B
        int[] b = new int[n];

        // Populate B: b[a[i] - 1] = i
        for (int i = 0; i < n; i++) {
            // a[i] is 1-based value, a[i] - 1 is 0-based index
            b[a[i] - 1] = i;
        }

        // Calculate the sum of absolute differences
        long res = 0;
        for (int i = 1; i < n; i++) {
            res += Math.abs(b[i] - b[i - 1]);
        }

        System.out.println(res);
        
        scanner.close();
    }
}

File: output/gemma4-e4b/avatar/Java/Python/codeforces_612_B.py
import sys

def solve():
    # Read all input tokens from stdin
    data = sys.stdin.read().split()
    if not data:
        return

    # N is the first element
    N = int(data[0])
    
    # The next N elements are the values A_i
    A = [int(x) for x in data[1:N+1]]
    
    # dp array initialization. Size 200001 is used in Java.
    # We initialize it to a size large enough to hold indices up to N (or 200000).
    MAX_SIZE = 200001
    dp = [0] * MAX_SIZE
    
    # Populate dp: A[j] is the value, j is the position (0-indexed)
    for j in range(N):
        value = A[j]
        # dp[value] = j
        if value < MAX_SIZE:
            dp[value] = j
            
    # Calculate the answer
    ans = 0
    # Loop from i = 2 up to N (inclusive)
    for i in range(2, N + 1):
        # ans += Math.abs(dp[i] - dp[i - 1])
        ans += abs(dp[i] - dp[i - 1])
        
    print(ans)

solve()